File: shop/src/main/java/basic/shop/common/model/Money.java
package basic.shop.common.model;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString(of = {"value"})
@AllArgsConstructor
public class Money {
    private int value;

    public int getValue() {
        return value;
    }

    @Override
    public boolean equals(Object o) {
        if(o == this)
            return true;
        if(!(o instanceof Money))
            return false;
        Money money = (Money) o;
        return value == money.value;
    }

    @Override
    public int hashCode() {
        return Integer.hashCode(value);
    }
}

File: spring-mvc-with-kotlin/order-api/src/main/kotlin/basic/orderapi/common/model/Money.kt
package basic.orderapi.common.model

class Money (
    var value: Int
) {
    fun multiply(multiplier: Int): Money {
        return Money(value * multiplier)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Money

        if (value != other.value) return false

        return true
    }

    override fun hashCode(): Int {
        return value
    }


}